### DISCUSSION TITLE
DIfferent cache configuration for different cache name with Springboot autoconfiguration

### DISCUSSION THREAD

**Discussion by nehrump** [ben-manes/caffeine#985] [Q&A] [answered]
Is there any way to define different cache configurations with different cache names ?
Currently as per my understanding, We can define multiple caches with application.yml alone but has only global cache configuration for all cache names.
I would like to define different set of caches with respective cache configuration with Spring boot autoconfiguration.
 Is there a solution for this already or Can I expect the solution in near future ?
Thanks in advance !!!

**Comment by ben-manes** [comment]
According to this [tutorial](https://www.youtube.com/watch?v=ze3qFp5pGeA&t=270s) you can use cache names. Does that work?

<img width="315" alt="Screen Shot 2023-05-29 at 11 09 33 PM" src="https://github.com/ben-manes/caffeine/assets/378614/7f83b076-eaa4-4476-b10e-0de30992ac83">

**Reply by nehrump** [reply]
I tried that way. But it is not working. Is there any particular caffeine or spring version required for this to work?

**Reply by ben-manes** [reply]
I don’t know, but before that there was an [alternative provider](https://github.com/stepio/coffee-boots). You’d need to ask the Spring folks.

**Reply by ben-manes** [reply, marked as answer]
I haven’t used Spring since it’s xml days, 2.1 iirc. The Spring support was a rename of their guava support, which was then depreciated and removed. How it functions was designed and supported by them, so unfortunately you’ll need to ask Spring experts for help. I think in past discussions they recommended the Java configuration, but I don’t know enough to explain why.
